bears/general/FilenameBear.py

```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        if (new_name != filename_without_extension or
+            not filename_without_extension.startswith(filename_prefix) or
+                not filename_without_extension.endswith(filename_suffix)):
+            correct_prefix_name = (new_name
+                                   if
+                                   filename_without_extension.startswith(
+                                                               filename_prefix)
+                                   else
+                                   filename_prefix + new_name)
+
+            correct_prefixsuffix_name = (correct_prefix_name
+                                         if
+                                         filename_without_extension.endswith(
+                                                               filename_suffix)
+                                         else
+                                         correct_prefix_name + filename_suffix)
```

Comment:
so you would insert something like this:
```
if not filename_without_extension.startswith(prefix):
    # modify new_name accordingly how it should be.

# same with suffix
```
Then you don't have to change any code in the result, except the message handling. Actually you can append messages inside a message list:

```
if not filename_without_extension.startswith(prefix):
    # modify new_name accordingly how it should be.
    messages.append('my message')
```
Then you don't need `itertools.compress` btw :)